lib/dpl/provider/pages.rb

```diff
@@ -17,23 +21,38 @@ class Pages < Provider
 
       require 'tmpdir'
 
+      if RUBY_VERSION >= "2.0.0"
+        requires 'octokit', version: '~> 4.6.2'
+      else
+        requires 'octokit', version: '~> 4.3.0'
+      end
+
       experimental 'GitHub Pages'
 
       def initialize(context, options)
         super
 
-        @build_dir = options[:local_dir] || '.'
+        @build_dir = File.join(Dir.pwd, options[:local_dir] || '.')
         @project_name = options[:project_name] || fqdn || slug
         @target_branch = options[:target_branch] || 'gh-pages'
 
         @gh_fqdn = fqdn
         @gh_url = options[:github_url] || 'github.com'
         @gh_token = option(:github_token)
+        @keep_history = !!keep_history
+        @allow_empty_commit = !!allow_empty_commit
+        @committer_from_gh = !!committer_from_gh
+        @verbose = !!verbose
 
         @gh_email = options[:email] || 'deploy@travis-ci.org'
         @gh_name = "#{options[:name] || 'Deployment Bot'} (from Travis CI)"
 
         @gh_ref = "#{@gh_url}/#{slug}.git"
+        @gh_remote_url = "https://#{@gh_token}@#{@gh_ref}"
```

Comment:
(for future)
We could reuse [`~/.netrc` as described here](https://docs.travis-ci.com/user/private-dependencies/#API-Token) as an additional shield against token exposal to log

Reply:
Unfortunately, git doesn't support bypassing `--netrc-file` argument to underlying `curl` and I don't want to modify user's `~/.netrc`